### ISSUE TITLE
passwords of length 32 characters are not properly passed when authenticating

### ISSUE THREAD

**Issue by blitzd** [pymssql/pymssql#325]
I did some troubleshooting on this for quite a while, I'm able to connect fine on linux with tsql, but pymssql keeps throwing an adaptive server connection issue. Our passwords are standardized at 32 characters, upper and lower case letters with numbers (and nothing else) - this will not work with pymssql, with the SQL Server logging 'Password did not match that for the login provided'. Cutting down the password size to 30 works fine, but 31+ characters will cause an error.

**Comment by msabramo** [comment]
Yes, I ran into this once before and it turned out to be a FreeTDS issue (pymssql is a fairly thin wrapper around FreeTDS). I can't remember if it was a bug that was fixed in FreeTDS or if it has to do with what TDS version you connect with. IIRC, older versions of the TDS protocol had a 30 character limit, but newer versions like 7.1 relaxed that restriction.

**Comment by msabramo** [comment]
See:

- http://lists.ibiblio.org/pipermail/freetds/2013q4/028705.html
- https://gitorious.org/freetds/freetds/commit/74e5457faef4d0116b3e8f6944790424efa74890

Try upgrading the version of FreeTDS.

**Comment by FlipperPA** [comment]
Here's an odd situation. I had a 25-character password that works fine with tsql, pyodbc, and directly connecting on SSMS to SQL Server 2014. It fails to work for pymssql with "Adaptive Connection Failed". I reduced the password to 16 characters and it connects successfully.

**Comment by msabramo** [comment]
@FlipperPA: Were the characters that you removed "special" - e.g.: non-ASCII?

You might try upgrading your FreeTDS as well.

**Comment by FlipperPA** [comment]
They were not special - all alphanumeric, A-Za-Z0-9. While I'd prefer to be on FreeTDS 0.95, instead of 0.91, how does pyodbc work with the same 25-character password over FreeTDS? I'll dig into the code a bit and see if I can find anything. I've run into this before (on CentOS, for what its worth).

**Comment by rsyring** [comment]
pyodbc and pymssql use completely different FreeTDS internal APIs. It's 
very possible something could be working in pyodbc and not in pymssql 
for the same FreeTDS version that is a result of FreeTDS itself and not 
a bug in the python libraries.

I can't say for sure that is the case here, but it's very possible from 
a FreeTDS APIs perspective.

*Randy Syring*
Chief Executive Developer
Direct: 502.276.0459
Office: 812.285.8766
Level 12 <https://www.level12.io/>
